jurigged/codetools.py

```diff
@@ -658,14 +685,31 @@ class FunctionDefinition(GroupDefinition):
     # Management #
     ##############
 
-    def stash(self, lineno=1, col_offset=0):
+    def _stash(self, lineno=1, col_offset=0):
+        # Original jurigged stash logic
+
         if not isinstance(self.parent, FunctionDefinition):
             co = self.get_object()
             if co and (delta := lineno - co.co_firstlineno):
                 self.recode(shift_lineno(co, delta), use_cache=False)
 
         return super().stash(lineno, col_offset)
 
+    def stash(self, lineno=1, col_offset=0):
+        # patch: There's an off-by-one bug coming from somewhere in jurigged.
+        #              This affects replaced functions. When line numbers are wrong
+        #              the debugger and inspection logic doesn't work as expected.
+        # TODO: We should fix this in jurigged itself, but porting over from here for now:
+        # https://github.com/breuleux/jurigged/issues/29
```

Comment:
IIRC there's some places where the line number is tweaked. It would make sense to fix it there, but I would want @breuleux to comment

Reply:
As per https://github.com/breuleux/jurigged/issues/29#issuecomment-2053891426 this was likely compensating for an issue with decorators and we can remove this since it wasn't a proper fix anyways